src/tests/ilverify/TestDataLoader.cs

```diff
@@ -176,20 +176,37 @@ private static TheoryData<TestCase> GetTestMethodsFromDll(Func<string[], MethodD
                     var method = (EcmaMethod)testModule.GetMethod(methodHandle);
                     var methodName = method.Name;
 
-                    if (!String.IsNullOrEmpty(methodName) && methodName.Contains("_"))
+                    if (methodName?.Contains("_") == false)
+                        continue;
+
+                    var index = methodName.LastIndexOf("_Valid");
+                    if (index < 0)
+                        index = methodName.LastIndexOf("_Invalid");
+                    if (index < 0)
+                        continue;
+
+                    var substring = methodName.Substring(index + 1);
+                    var split = substring.Split('_');
+                    string[] mparams = new string[split.Length + 1];
+                    split.CopyTo(mparams, 1);
+                    mparams[0] = methodName.Substring(0, index);
+                    // examples of methodName to mparams transformation:
+                    //   * `get_Property` -> [ 'get_Property' ]
+                    //   * `CheckSomething_Valid` -> [ 'CheckSomething', 'Valid' ]
+                    //   * 'WrongMethod_Invalid_BranchOutOfTry' -> [ 'WrongMethod', 'Invalid', 'BranchOutOfTry' ]
+                    //   * 'MoreWrongMethod_Invalid_TypeAccess.InitOnly' -> [ 'MoreWrongMethod', 'Invalid', 'TypeAccess', 'InitOnly' ]
```

Comment:
> TypeAccess.InitOnly

Should this be `TypeAccess_InitOnly`? I do not see where we are recognizing `.` as the separator.

Reply:
We have 25 cases using period as a error code separator when there is more than one error:

```sh
$ git grep '_Invalid_.*\..*\(.*\)' ':/src/tests/ilverify/ILTests'
```
It is documented here https://github.com/dotnet/runtime/tree/main/src/coreclr/tools/ILVerify#methods-with-special-names. I agree it is quite complicated; imo, we should use xml or json list for this rather than complex naming convention parsing.

Reply:
> we should use xml or json list

Then you have a problem with the IL and json being out-of-sync all the time. I actually like the convention based scheme we have here.

Reply:
> Then you have a problem with the IL and json being out-of-sync all the time.

Current approach is equally fragile. The type discovery does not give any indication whether or not it has skipped the test during parsing (which is spread across three separate methods of TestDataLoader.cs). We have to remember total count before and after the new additions.